mysql-test/suite/tianmu/t/issue538.test

```diff
@@ -0,0 +1,132 @@
+
+--echo #
+--echo # issue 538 test for user defined function of tianmu
+--echo #
+
+use test;
+
+--disable_warnings
+
+DROP TABLE IF EXISTS employees;
+
+CREATE TABLE `employees` (
+  `employee_id` int(11) NOT NULL ,
+  `employee_name` varchar(50) NOT NULL,
+  `employee_sex` varchar(10) DEFAULT 'men',
+  `hire_date` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `employee_mgr` int(11) DEFAULT NULL,
+  `employee_salary` float DEFAULT '3000',
+  `department_id` int(11) DEFAULT NULL
+) ENGINE=TIANMU DEFAULT CHARSET=utf8mb4;
+
+DROP FUNCTION IF EXISTS get_desc;
+DROP FUNCTION IF EXISTS get_int;
+
+CREATE FUNCTION get_desc(id INT) RETURNS VARCHAR(300) 
+  RETURN (SELECT CONCAT('employee name:',employee_name,'---','salary: ',employee_salary) FROM employees WHERE employee_id=id);
+
+CREATE FUNCTION get_int(id INT) RETURNS VARCHAR(300) 
+  RETURN (SELECT employee_salary FROM employees WHERE employee_id=id);
+
+--disable_query_log
+
+INSERT INTO employees(employee_id,employee_name,employee_sex,employee_mgr,employee_salary,department_id) VALUES (1,'David Tian','man',10,7500,1);
+INSERT INTO employees(employee_id,employee_name,employee_sex,employee_mgr,employee_salary,department_id) VALUES (2,'Black Xie','man',10,6600,1);
+INSERT INTO employees(employee_id,employee_name,employee_sex,employee_mgr,employee_salary,department_id) VALUES (3,'Moses Wang','man',10,4300,1);
+INSERT INTO employees(employee_id,employee_name,employee_sex,employee_mgr,employee_salary,department_id) VALUES (4,'Rena Ruan','woman',10,5300,1);
+INSERT INTO employees(employee_id,employee_name,employee_sex,employee_mgr,employee_salary,department_id) VALUES (5,'Sunshine Ma','woman',10,6500,2);
+
+--enable_query_log
+
+SELECT
+	a.employee_id,
+	get_desc(b.employee_id)
+FROM
+	employees a
+LEFT JOIN employees b ON
+	a.employee_id = b.employee_id
+	and b.employee_name = 'David Tian';
+
+SELECT
+	sum(200 + get_int(b.employee_id))
+FROM
+	employees a
+LEFT JOIN employees b ON
+	a.employee_id = b.employee_id
+	and b.employee_name = 'David Tian';
+
+SELECT
+	a.employee_id,
+	get_desc(b.employee_id)
+FROM
+	employees a
+LEFT JOIN employees b ON
+	a.employee_salary = get_int(b.employee_id)
+	and b.employee_name = 'David Tian';
+
+SELECT
+	a.employee_id,
+	get_desc(b.employee_id)
+FROM
+	employees a
+LEFT JOIN employees b ON
+	b.employee_name = 'David Tian'
+WHERE
+	a.employee_salary < get_int(b.employee_id);
+
+SELECT
+	a.employee_id,
+	get_desc(b.employee_id)
+FROM
+	employees a
+LEFT JOIN employees b ON
+	b.employee_name = 'David Tian'
+WHERE
+	a.employee_salary > get_int(b.employee_id);
+
+SELECT
+	a.employee_id,
+	get_desc(b.employee_id)
+FROM
+	employees a
+LEFT JOIN employees b ON
+	b.employee_name = 'David Tian'
+WHERE
+	a.employee_salary = get_int(b.employee_id);
+
+SELECT
+    a.employee_id,
+    get_desc(b.employee_id)
+FROM
+    employees a
+LEFT JOIN employees b ON
+    b.employee_name = 'David Tian'
+WHERE
+    get_int(b.employee_id) = a.employee_salary;
```

Comment:
Nice job. 
BTW:  pls, add tests for like as following:

a udf may return an array or..
```
select xxx from udf(xxx) as tb.... 
```